tests.sh

```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 
-cd tests/binaries && make && cd ../..
+cd ./tests/binaries || exit 1
+make clean all || exit 2
```

Comment:
The reason of some failures in the tests is that it take the old assembled and linked files, to improve this task it's better to generate the files every time we run the tests. If `make` is executed but the files `.out` existed, then it won't create any new file, for this reason, is better to clean and re-build.